Mathlib/MeasureTheory/Measure/WithDensity.lean

```diff
@@ -379,13 +386,57 @@ theorem set_lintegral_withDensity_eq_set_lintegral_mul_non_measurable₀ (μ : M
   rw [restrict_withDensity hs, lintegral_withDensity_eq_lintegral_mul_non_measurable₀ _ hf h'f]
 #align measure_theory.set_lintegral_with_density_eq_set_lintegral_mul_non_measurable₀ MeasureTheory.set_lintegral_withDensity_eq_set_lintegral_mul_non_measurable₀
 
-theorem withDensity_mul (μ : Measure α) {f g : α → ℝ≥0∞} (hf : Measurable f) (hg : Measurable g) :
+theorem withDensity_mul₀ {μ : Measure α} {f g : α → ℝ≥0∞}
+    (hf : AEMeasurable f μ) (hg : AEMeasurable g μ) :
     μ.withDensity (f * g) = (μ.withDensity f).withDensity g := by
   ext1 s hs
-  simp [withDensity_apply _ hs, restrict_withDensity hs,
-    lintegral_withDensity_eq_lintegral_mul _ hf hg]
+  rw [withDensity_apply _ hs, withDensity_apply _ hs, restrict_withDensity hs,
+    lintegral_withDensity_eq_lintegral_mul₀ hf.restrict hg.restrict]
+
+theorem withDensity_mul (μ : Measure α) {f g : α → ℝ≥0∞} (hf : Measurable f) (hg : Measurable g) :
+    μ.withDensity (f * g) = (μ.withDensity f).withDensity g :=
+  withDensity_mul₀ hf.aemeasurable hg.aemeasurable
 #align measure_theory.with_density_mul MeasureTheory.withDensity_mul
 
+lemma withDensity_inv_same_le {μ : Measure α} {f : α → ℝ≥0∞} (hf : AEMeasurable f μ) :
```

Comment:
This one is probably true without `hf`, but it's probably not worth it since the proof will be more painful while we will only use your version.